crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -38,11 +39,34 @@ pub fn get_property(
     }
 }
 
+pub fn get_scarb_manifest(path: &Option<Utf8PathBuf>) -> Result<Utf8PathBuf> {
+    which::which("scarb")
+        .context("Cannot find `scarb` binary in PATH. Make sure you have Scarb installed https://github.com/software-mansion/scarb")?;
+
+    if let Some(path) = path {
+        return Ok(path.clone());
+    }
+
+    let output = Command::new("scarb")
+        .arg("manifest-path")
+        .stdout(Stdio::piped())
+        .output()
+        .context("Failed to execute scarb manifest-path command")?;
+
+    let output_str = String::from_utf8(output.stdout)
+        .context("Invalid output of scarb manifest-path command")?;
+
+    let path = Utf8PathBuf::from_str(output_str.trim())
+        .context("Scarb manifest-path returned invalid path")?;
+
+    Ok(path)
+}
+
 pub fn parse_scarb_config(
     profile: &Option<String>,
     path: &Option<Utf8PathBuf>,
 ) -> Result<CastConfig> {
-    let found_path = find_manifest_path(path.as_ref().map(Utf8PathBuf::as_path));
+    let found_path = get_scarb_manifest(path);
```

Comment:
```suggestion
    let found_path = path.unwrap_or_else(|| get_scarb_manifest());
```
Seems more explicit to me, as you are not using the path in `get_scarb_manifest` anyway

Reply:
Makes sense, but I also like the way `get_scarb_manifest` checks for scarb existence in PATH (and it is used everywhere where scarb is required). 

By applying you suggestion, I would have to copy this part into multiple places in my code, because it wouldn't be invoked when path to scarb.toml is supplied, right? Also, this is kind of inline with scarb's own `get_manifest_path` where you can override the path, so I thought this is ok to do something similar.

Reply:
> Makes sense, but I also like the way get_scarb_manifest checks for scarb existence in PATH (and it is used everywhere where scarb is required).

I suggest not calling `scarb manifest-path` each time we need to use Scarb, as it often can be redundant. Which makes uncoupling it from ensuring Scarb is installed sensible for me anyway.

> Also, this is kind of inline with scarb's own get_manifest_path where you can override the path, so I thought this is ok to do something similar.

It's probably ok :D
 However, contrary to this function, `find_manifest_path` from Scarb does actually use the variable passed to it. If the variable is not used, I see passing it to the function as unnecessary cognitive complexity.

Reply:
> I suggest not calling scarb manifest-path each time we need to use Scarb, as it often can be redundant. Which makes uncoupling it from ensuring Scarb is installed sensible for me anyway.

Well, if path is supplied it is not called at all. If it's not supplied it is called at most 2 times, and I agree that in this scenario this could be scaled down to 1. I'll try to make that happend in this PR.

> It's probably ok :D
However, contrary to this function, find_manifest_path from Scarb does actually use the variable passed to it. If the variable is not used, I see passing it to the function as unnecessary cognitive complexity.

I want to understand this better, so sorry for silly question - my understanding is that find_manifest_path from Scarb checks for path in cwd or parent dirs if it is supplied with 'None' argument; otherwise it just returns the supplied path, right? So I suppose it does nothing more with the supplied path than get_scarb_manifest here in cast - when supplied with None it searches for path, otherwise it returns it. Or am I missing something?

Thanks for all the tips though! :)

Reply:
```
        Some(user_override) => Ok(fs::canonicalize(user_override)
            .unwrap_or_else(|_| user_override.into())
            .try_into_utf8()?),
```
Not much, but it is canonicalized and parsed there